internal/blockdb/migrate.go

```diff
@@ -159,3 +180,12 @@ FROM v_tx_flattened, json_each(v_tx_flattened.tx, "$.body.messages")
 
 	return nil
 }
+
+func errIgnoreDuplicateColumn(err error, col string) error {
+	var serr *sqlite.Error
+	if errors.As(err, &serr) &&
```

Comment:
`&serr` doesn't need to be dereferenced here. It looks like everything is working as intended, but this is one corner of Go that I don't understand -- why is the address to a pointer accepted?

Reply:
Github isn't letting me comment on your previous comment. ¯\_(ツ)_/¯

> Moreover, we don't yet have proper migration tests in place. 

This is hard to pull off in practice. I've never worked at a place that did this well if at all. There's always bad and surprising data that you never accounted for. At previous companies, the strategy was if migrations failed, you did a quick rollback of the application. But that's in the classic web server context; it's different for embedded databases.

In this case, I think committing to the strategy of we delete (or backup) and start over is the way to go. 

I'll attempt a PR at ☝️ strategy first.

Reply:
> &serr doesn't need to be dereferenced here.

It does need to be dereferenced, otherwise my IDE prints the warning: `Second argument to 'errors.As' must be a pointer to an interface or to a type implementing the error interface`. 

Only the pointer type to `sqlite.Error` implements the error interface (not the value).